ql/experimental/volatility/noarbsabrinterpolation.hpp

```diff
@@ -217,24 +217,24 @@ class NoArbSabrInterpolation : public Interpolation {
                 {alphaIsFixed, betaIsFixed, nuIsFixed, rhoIsFixed},
                 vegaWeighted, endCriteria, optMethod, errorAccept, useMaxError,
                 maxGuesses));
-        coeffs_ = ext::dynamic_pointer_cast<
-            detail::XABRCoeffHolder<detail::NoArbSabrSpecs> >(impl_);
```

Comment:
`coeffs_` is the member being sliced off when this class is copied by value into an instance of `Interpolation`, because it lives within the layout of `NoArbSabrInterpolation`, not `Interpolation`. The exact warning I get is the following:

`[build] /home/sweemer/QuantLib/ql/experimental/volatility/noarbsabrinterpolation.hpp:267:16: warning: slicing object from type 'NoArbSabrInterpolation' to 'Interpolation' discards 16 bytes of state [cppcoreguidelines-slicing]`

The way to prevent slicing is to do the dynamic pointer cast on `impl_`, which does live inside of `Interpolation`'s layout.

Reply:
Oh, I see, these lines cause the warning but it's emitted elsewhere.  Ok, I'm sold.